docs/source/using-federation/api/apollo-gateway.mdx

```diff
@@ -0,0 +1,825 @@
+
+When called, the gateway sends a small query (`{ __typename }`) to each subgraph to verify that they're all responsive. This function `throw`s on failure and returns a `Promise` to be `await`ed.
+
+##### Parameters
+
+<table class="field-table">
+  <thead>
+    <tr>
+      <th>Name /<br/>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+<tr>
+<td>
+
+###### `serviceMap`
+
+`Object` (`DataSourceMap`)
+
+</td>
+<td>
+
+If provided, the gateway sends health check requests to only the data sources included in the map.
+
+By default, the gateway uses `this.serviceMap` (i.e., it sends health check requests to all known subgraphs).
+
+</td>
+</tr>
+</tbody>
+</table>
+
+## `class RemoteGraphQLDataSource`
+
+Represents a connection between your federated gateway and _one_ of your subgraphs.
+
+You can customize this connection by extending this class and overriding its `willSendRequest` and/or `didReceiveResponse` methods:
+
+- Override `willSendRequest` to modify your gateway's requests to the subgraph before they're sent.
+- Override `didReceiveResponse` to modify the subgraph's responses before the gateway passes them along to the requesting client.
+
+These methods are described in detail below.
+
+### Methods
+
+#### `constructor`
+
+Returns an initialized `RemoteGraphQLDataSource` instance:
+
+<MultiCodeBlock>
+
+```ts
+const productsDataSource = new RemoteGraphQLDataSource({
+  url: 'https://products-service.dev/graphql',
+});
+```
+
+</MultiCodeBlock>
+
+Takes an `options` object as a parameter. Supported fields of this object are described below.
+
+##### Options
+
+<table class="field-table">
+  <thead>
+    <tr>
+      <th>Name /<br/>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+<tr class="required">
+<td>
+
+###### `url`
+
+`String`
+
+</td>
+<td>
+
+**Required.** The subgraph's URL for sending fetch requests via HTTP.
+
+</td>
+</tr>
+
+<tr>
+<td>
+
+###### `apq`
+
+`Boolean`
+
+</td>
+<td>
+
+If `true`, the gateway attempts to use [automated persisted queries (APQ)](../../performance/apq/) when sending queries to this subgraph. APQ can dramatically reduce the size of most requests sent over the network, especially for more complex queries.
+
+The subgraph must also enable support for APQ for the gateway to use this feature (Apollo Server enables APQ by default).
+
+</td>
+</tr>
+</tbody>
+</table>
+
+#### `willSendRequest`
+
+Override this method in a subclass to modify each outgoing fetch request before it's sent to the subgraph:
+
+<MultiCodeBlock>
+
+```ts
+// Adds an `x-user-id` header to each outgoing fetch request
+class AuthenticatedDataSource extends RemoteGraphQLDataSource({
+  willSendRequest({ request, context }) {
+    request.http.headers.set('x-user-id', context.userId);
+  }
+});
+```
+
+</MultiCodeBlock>
+
+This method takes a `requestContext` object that contains both the original unmodified `request` and the current `context` object (i.e., the `contextValue` object).
```

Comment:
This article is mostly the same **except** for a few references to `contextValue` that I added. [Once the context article is merged in](https://github.com/apollographql/apollo-server/pull/7182/), I'll also add accompanying links which should help make this even clearer.